aten/src/ATen/native/README.md

```diff
@@ -286,6 +286,32 @@ In addition to the backends above, we also support the keywords:
     registered to `Math` key should be plain mathematical composition of other
     `at::` functions and support training and inference for any backend.
 
+Note the kernels registered to `DefaultBackend` and `Math` might look identical,
+they only differ in whether it's also used to compute backward as well.
+
+For example, suppose `my_op` can be decomposed in the following way:
+```
+at::Tensor my_op(const Tensor& self, const Tensor& other) {
+  return self + 2 * other;
+}
+```
+
+If we already know inference kernels and derivative formulas for operators `+` and `*` in our system,
+you can just register `my_op` to `Math` and both inference & autograd will just work.
+Although it seems we only write down the inference formula here, PyTorch autograd system would correctly
+set up the backward for `my_op` using the chain formula and derivatives of `+` & `-` operators.
+In other words `d_out/d_self = 1; d_out/d_other = 2` can be derived automatically from
+`my_op` inference kernel. Of course if we don't have derivative formula defined for any of `+` or `-`,
+backward of `my_op` can no longer be derived automatically.
+
+Whether to use `Math` or `DefaultBackend` for your kernel can be decided by the following steps:
+1. Always start with a `Math` kernel that's composable of other existing operators.
+2. If you don't want to use the derived gradient formula from `Math` kernel for autograd, either to
+   get better performance or better numerical stability, you should put the kernel in `DefaultBackend`
+   so that it's only used in inference.
+   Later for autograd, depending on whether your autograd kernel works for all backends or not,
+   you can put them in alias `Autograd` or specific keys like `AutogradCPU`.
+
 If you add `dispatch` section to any API that didn't have it before, you **have to** move
 the old implementation to `Math` field so that it's still available for other backends to use.
```

Comment:
I think we need to clarify what we're telling them to do here. "move the old implementation to the `Math` field" means specifying a kernel name that wasn't there before, right? The name we generate is mostly identical to the op name, but e.g. we need to tell what to do about overload names (I think we put an underscore between the base name and the overload name, but I'm not sure) and whatever other name mangling we do.

Aside: I notice there are some ops in `native_functions.yaml` that only specify `Math` kernels right now, with the same name as the op (random examples: `broadcast_to`, `row_stack`. Are those entries necessary?